pkg/assisted/agentserviceconfig_test.go

```diff
@@ -0,0 +1,810 @@
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakeRuntimeClient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	agentInstallV1Beta1 "github.com/openshift/assisted-service/api/v1beta1"
+)
+
+func TestNewAgentServiceConfigBuilder(t *testing.T) {
+	testCases := []struct {
+		databaseStorageSize   resource.Quantity
+		filesystemStorageSize resource.Quantity
+		databaseStorage       corev1.PersistentVolumeClaimSpec
+		filesystemStorage     corev1.PersistentVolumeClaimSpec
+		client                bool
+	}{
+		{
+			databaseStorageSize:   resource.MustParse("50Gi"),
+			filesystemStorageSize: resource.MustParse("10Gi"),
+			databaseStorage: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("50Gi"),
+					},
+				},
+			},
+			filesystemStorage: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("10Gi"),
+					},
+				},
+			},
+			client: true,
+		},
+		{
+			databaseStorageSize:   resource.MustParse("50Gi"),
+			filesystemStorageSize: resource.MustParse("10Gi"),
+			databaseStorage: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("50Gi"),
+					},
+				},
+			},
+			filesystemStorage: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("10Gi"),
+					},
+				},
+			},
+			client: false,
+		},
+	}
+
+	for _, testcase := range testCases {
+		var testSettings *clients.Settings
+		if testcase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{})
+		}
+
+		testBuilder := NewAgentServiceConfigBuilder(
+			testSettings, testcase.databaseStorage, testcase.filesystemStorage)
+
+		if testcase.client {
+			assert.Equal(t, testcase.databaseStorageSize,
+				*testBuilder.Definition.Spec.DatabaseStorage.Resources.Requests.Storage())
+			assert.Equal(t, testcase.filesystemStorageSize,
+				*testBuilder.Definition.Spec.FileSystemStorage.Resources.Requests.Storage())
+			assert.Empty(t, testBuilder.errorMsg)
+		} else {
+			assert.Nil(t, testBuilder)
```

Comment:
After you mutate your functions please check testBuilder.errorMsg

Reply:
The negative scenario is when supplying empty apiClient to new builder. In this case testBuilder is nil so errorMsg dose not exist